osbuild/objectstore.py

```diff
@@ -22,6 +23,70 @@ class Mode(enum.Enum):
         READ = 0
         WRITE = 1
 
+    class Metadata:
+        """store and retrieve metadata for an object"""
+
+        def __init__(self, path) -> None:
+            self.path = path
+            os.makedirs(path, exist_ok=True)
+
+        def _path_for_key(self, key) -> str:
+            assert key
+            name = f"{key}.json"
+            return os.path.join(self.path, name)
+
+        @contextlib.contextmanager
+        def write(self, key):
+
+            tmp = tempfile.NamedTemporaryFile(
+                mode="w",
+                encoding="utf8",
+                dir=self.path,
+                prefix=".",
+                suffix=".tmp.json",
+                delete=True,
+            )
```

Comment:
With `FsCache` you wouldn't need any temporaries, since any modifications would always happen in the staging-area, and this gets discarded if you don't commit it. With the old cache, We might also be fine without temporaries, since we don't set the `refs/` link if we fail, don't we?

Reply:
Yeah, it technically doesn't need to be a temp file. The behaviour I want is that if there is nothing in the file it gets deleted.